.circleci/config.yml

```diff
@@ -1,240 +1,312 @@
     steps:
-      - checkout
-      - run: |
-          sed -i '/<TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0<\/TargetFrameworks>/c\<TargetFramework>'net7.0'<\/TargetFramework>' Client.Core.Test/Client.Core.Test.csproj
-          sed -i '/<TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0<\/TargetFrameworks>/c\<TargetFramework>'net7.0'<\/TargetFramework>' Client.Test/Client.Test.csproj
-          sed -i '/<TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0<\/TargetFrameworks>/c\<TargetFramework>'net7.0'<\/TargetFramework>' Client.Legacy.Test/Client.Legacy.Test.csproj
-          sed -i '/<TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0<\/TargetFrameworks>/c\<TargetFramework>'net7.0'<\/TargetFramework>' Client.Linq.Test/Client.Linq.Test.csproj
-          sed -i '/<TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0<\/TargetFrameworks>/c\<TargetFramework>'net7.0'<\/TargetFramework>' Examples/Examples.csproj
-      - run:
-          name: Check compilation warnings
-          command: |
-            dotnet clean --configuration Release
-            dotnet build --configuration Release -warnAsError -warnAsMessage:CS0618,NETSDK1138
-      
+      - attach_workspace:
+          at: .
+      - influxdb_install_windows:
+          zip-file-name: influxdb2-2.7.5-windows.zip
+          influxdb-version: 2.7.5
+      - influxdb_install_windows:
+          zip-file-name: influxdb-1.8.10_windows_amd64.zip
+          influxdb-version: 1.8.10
+      - influxdb_run_windows:
+          influxdb-version: 2.7.5
+          influxdb-port: 9999
+          influxdb-data-dir: ~/.influxdbv2
+      - influxdb_run_windows:
+          influxdb-version: 2.7.5
+          influxdb-port: 8888
+          influxdb-data-dir: ~/.influxdbv2-onboarding-tests
+      - influxdb_run_windows:
+          influxdb-version: 1.8.10
+          influxdb-port: 8086
+          influxdb-data-dir: ~/.influxdbv1
+      - dotnet_info
+      - test_influxdb_client:
+          dotnet-target-framework: << parameters.dotnet-target-framework >>
+          code-coverage-report: << parameters.code-coverage-report >>
+
   check-code-formatting:
     docker:
       - image: *default-dotnet-image
+    working_directory: ~/influxdb-client-csharp
     steps:
       - checkout
       - restore_cache:
-          name: Restoring reSharper Cache
+          name: "Restore ReSharper cache"
           keys:
-              - &cache-key reSharper-cache-2022_1_0
+            - &cache-key ReSharper-cache-2022_1_0
       - run:
-          name: Check code formatting
+          name: "Check code formatting"
+          shell: bash
           command: |
             ./Scripts/reformat-code.sh
             git diff --exit-code
       - save_cache:
-            name: Saving reSharper Cache
-            key: *cache-key
-            paths:
-                - ./reSharperCLI
-  
+          name: "Save ReSharper cache"
+          key: *cache-key
+          paths:
+            - ./ReSharperCLI
+
   deploy-preview:
```

Comment:
How do you deploy production NuGet packages (those without `-dev.CIRCLE_BUILD_NUM`)?
We can probably integrate this deployment phase in this pipeline with a [manual approval](https://circleci.com/docs/workflows/#holding-a-workflow-for-a-manual-approval).

Reply:
We don't want to deploy production packages by CI.

Reply:
Ok, that's a shame because it would keep track of build and test logs, and would also guarantee that all tests are carried out on the exact same assemblies deployed in production...